packages/thrift-server-core/src/main/Int64.ts

```diff
@@ -258,32 +261,45 @@ export class Int64 implements IInt64 {
     public static fromDecimalString(text: string): Int64 {
         const negative: boolean = text.charAt(0) === '-'
 
-        if (text.length < (negative ? 17 : 16)) {
-            // The magnitude is smaller than 2^53.
-            return new Int64(+text)
-        } else if (text.length > (negative ? 20 : 19)) {
+        // Short numbers are fast.
+        const numVal = parseInt(text, 10)
```

Comment:
I could leave this as just `+text` or `Number(text)`. The difference being that the whole string needs to be a number. However, the hex string handling already behaves more like `parseInt()`. Maybe that was a bad choice and it should have strictly required that the whole string be a hex value. I didn't realize the number conversion and `parseInt()` behaved that differently.